File: src/my_project/easy_problems/from201to250/k_distinct_string_array.py
from typing import List 

class Solution:
    def kthDistinct(self, arr: List[str], k: int) -> str:

        count_dict = dict()
        len_arr: int = len(arr)
        count: int = 0
        for i in range(len_arr):
            count_dict[arr[i]] = 0

        for i in range(len_arr):
            count_dict[arr[i]] += 1

        for i in count_dict.keys():
            if count_dict[i] == 1:
                count += 1
                if count == k:
                    return i 
        
        return "" 


solution = Solution()

print(solution.kthDistinct(arr = ["d","b","c","b","c","a"], k = 2))

print(solution.kthDistinct(arr = ["aaa","aa","a"], k = 1))

print(solution.kthDistinct(arr = ["a","b","a"], k = 3))

File: src/my_project_typescript/easy_problems_typescript/from101to150/kDistinctStringArray.ts
function kthDistinct(arr: string[], k: number): string {
    let countDict: any = {}
    let lenArr: number = arr.length
    let count: number = 0

    for (let i = 0; i< lenArr; i++){
        countDict[arr[i]] = 0
    }

    for (let i = 0; i< lenArr; i++){
        countDict[arr[i]] += 1
    }

    for (let i in countDict){
        
        if (countDict[i] == 1){
            count += 1
            if (count === k){
                return i 
            }
            
        }
    }
    return ""
};


console.log(kthDistinct(["d","b","c","b","c","a"],  2))

console.log(kthDistinct(["aaa","aa","a"],  1))

console.log(kthDistinct(["a","b","a"],  3))


